### ISSUE TITLE
Support for Omnitronic cmx2000 mixer

### ISSUE THREAD

**Issue by marcmiller1** [pideck/pideck#100]
Hi. I'm try by to write a file for my asoundrc.

I'm trying  set up a hybrid mixer, omnitronic cmx2000 to run 2 decks. I changed to umix44 file I found online to meet my needs, I having an issue that after the start up command I'm getting the message that my IPC key is incorrect for my sound card. Any ideas how to find the correct IPC. This is my first project so spoon feeding may be required!

**Comment by danielhjames** [comment]
Hi, ipc_key must be unique for each dmix configuration you create. Possibly you have reused this number from your umix44 configuration? Please see the options for dmix on https://www.alsa-project.org/alsa-doc/alsa-lib/pcm_plugins.html

**Comment by marcmiller1** [comment]
Thanks, got it working, 2 deck on the hybrid mixer! I'm like a kid at Xmas!

**Comment by danielhjames** [comment]
Thanks @marcmiller1 ! Please post your .asoundrc for this mixer and upload a video to YouTube if you can :-)

**Comment by marcmiller1** [comment]
One more question! Is it possible to add a load2 and cue 2
 I've had a look at the touchscreen settings but am struggling to see how to alter them

Thanks!

**Comment by danielhjames** [comment]
Yes, you can start the touchscreen controls with the `--card twindeck` option to get these extra buttons. Please see for example the way the Hercules Mk2 is set up in https://github.com/pideck/pideck/blob/master/cards_autodetect and add a similar stanza for your Omnitronic mixer.

You'll see that once a specific audio device is detected from the `/proc/asound/cards` file, xwax is set up with the variable XCARD which indicates the ALSA devices to use.

Then we copy any .asoundrc file required for this device into place in the user's home directory.

Finally we start the touchscreen buttons for left and right sides of the display with the appropriate `--card` argument.

**Comment by marcmiller1** [comment]
That's great I see how to sort it, but how do I access the file to enter the stanza for my mixer. 

Thanks again for all your help!!

**Comment by danielhjames** [comment]
On an installed system it should be something like /usr/share/pideck/cards_autodetect

Then when the /usr/share/pideck/pideck script is called on boot, it should pull in your updated version of cards_autodetect

**Comment by danielhjames** [comment]
![omnitroniccmx2000](https://user-images.githubusercontent.com/1867708/56136531-e1cdd780-5f8a-11e9-9c63-ccc90039aa86.jpg)

**Comment by marcmiller1** [comment]
I've tried to modify the Hercules set as you outlined above, and added the --card twindeck command to the startup, but nothing has changed, and clues? I changed the Hercules set up to "cmx-2000" as that's what my mixer sound card is referred to in the ~/.soundrc file I wrote. 

Thanks

**Comment by danielhjames** [comment]
Hi @marcmiller1 the soundcard detection may be case sensitive depending on the grep options used, so try using CMX2000 or CMX-2000 (in upper case) to match against, as per your screenshot above.

**Comment by marcmiller1** [comment]
Hi and thanks, I've got it working fully!! I did a bit of a dirty work around, and on the touchscreen set up I changed the last else if to include --card twindeck, changing the default to twindeck and all works great. Really appreciate all your help. I will upload a video on YouTube as soon as I get time. Thanks again. I've been using serato and vdj for years but now really love the simplicity of pideck.



Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by danielhjames** [comment]
Cool! Please send over your versions of the files you modified so we can merge these changes for the next release.
